src/components/modals/movieDetailsModal/movieDetailsModal.component.tsx

```diff
@@ -1,155 +1,129 @@
-      description: !movie.description,
-      duration: !movie.duration,
-    };
-
-    setValidationState(newValidationState);
-
-    return !Object.values(newValidationState).includes(true);
-  };
 
   return (
-    <div className={s.background}>
-      <h2 className={s.modalTitle}>
-        {modalDetails.type}
-        {' '}
-        movie
-      </h2>
-      <div className={s.scrollContainer}>
-        <form className={s.modalForm} autoComplete="off">
-          <label htmlFor="movieTitle">
-            title
-            <input
-              type="text"
-              id="movieTitle"
-              value={movie.title}
-              onChange={({ target: { value: title } }) => {
-                changeMovie({ ...movie, title });
-                setValidationState({ ...validationState, title: !title });
-              }}
-            />
-            {validationState.title && <p className={s.error}>please add the title.</p>}
-          </label>
-          <label htmlFor="movieYear">
-            release date
-            <input
-              type="date"
-              id="movieYear"
-              value={movie.releaseDate}
-              onChange={({ target: { value: releaseDate } }) => {
-                changeMovie({ ...movie, releaseDate });
-                setValidationState({ ...validationState, releaseDate: !releaseDate });
-              }}
-            />
-            {validationState.releaseDate
-              && (
-                <p className={s.error}>
-                  please choose a release date.
-                </p>
-              )}
-          </label>
-          <label htmlFor="movieUrl">
-            movie url
-            <input
-              type="url"
-              id="movieUrl"
-              value={movie.url}
-              onChange={({ target: { value: url } }) => {
-                changeMovie({ ...movie, url });
-                setValidationState({ ...validationState, url: !url });
-              }}
-            />
-            {validationState.url && <p className={s.error}>please add the url.</p>}
-          </label>
-          <label htmlFor="movieGenre">
-            genre
-            <MultipleDropdown
-              genres={movie.genre}
-              onGenreClick={(genre) => {
-                changeMovie({ ...movie, genre });
-                setValidationState({ ...validationState, genre: !genre.length });
-              }}
-            />
-            {validationState.genre && <p className={s.error}>please choose at least one genre.</p>}
-          </label>
-          <label htmlFor="movieDescription">
-            overview
-            <textarea
-              id="movieDescription"
-              value={movie.description}
-              onChange={({ target: { value: description } }) => {
-                changeMovie({ ...movie, description });
-                setValidationState({ ...validationState, description: !description });
-              }}
+    <Formik
+      initialValues={...movie}
```

Comment:
how does this syntax even works? 😮

Reply:
Object destructuring, probably :)

Reply:
but it should be destructed to another object, not a prop assignment, like `={{...movie}}`, writing something like `={...{ some: 'object'}}` fails the app, but this doesn't